nca/Parsers/IstioVirtualServiceYamlParser.py

```diff
@@ -0,0 +1,600 @@
+        if not result:
+            self.warning('no valid VirtualServices found. Ignoring istio ingress traffic')
+        else:
+            result[0].findings = self.warning_msgs + gtw_parser.warning_msgs
+        return result
+
+    def pick_vs_gateways_by_hosts(self, vs, gateway_names):
+        """
+        Given a list of gateway names, retrieves the gateways that match the virtual service by hosts
+        or gives a warning for non-existing or non-matching ones
+        :param VirtualService vs: the virtual service that references the gateways in the list
+        :param list[str] gateway_names: the given gateway names list
+        :return: dict: the map from gateways to the matching hosts.
+        """
+
+        gtw_to_hosts = {}
+        vs_all_hosts_dfa = reduce(MinDFA.__or__, vs.hosts_dfa)
+        for gtw_name in gateway_names:
+            if gtw_name == 'mesh':
+                continue
+            gtw = self.gtw_parser.get_gateway(gtw_name)
+            if gtw:
+                matching_hosts = gtw.all_hosts_dfa & vs_all_hosts_dfa
+                if matching_hosts:
+                    gtw_to_hosts[gtw] = matching_hosts
+                else:
+                    self.warning(f'Gateway {gtw_name}, referenced in the VirtualService {vs.full_name()} '
+                                 f'does not match to the VirtualService by hosts. Ignoring the gateway')
+            else:
+                self.warning(f'missing gateway {gtw_name}, referenced in the VirtualService {vs.full_name()}. '
+                             f'Ignoring the gateway')
+        return gtw_to_hosts
+
+    @staticmethod
+    def init_mesh_to_egress_policy(vs, selected_peers):
+        """
+        Initialization of GatewayPolicy for holding connections from mesh to egress gateway
+        :param VirtualService vs: the virtual service that defines the connections from mesh to egress gateway
+        :param PeerSet selected_peers: egress gateway pods
+        :return: the resulting GatewayPolicy
+        """
+        mesh_to_egress_policy = GatewayPolicy(vs.name + '/mesh/egress/allow', vs.namespace,
+                                              GatewayPolicy.ActionType.Allow)
+        mesh_to_egress_policy.policy_kind = NetworkPolicy.PolicyType.GatewayPolicy
+        # We model egress flow relatively to egress gateways pods (i.e. they are the selected_peers);
+        # since the flow is into those selected peers, the policy will affect ingress.
+        mesh_to_egress_policy.affects_ingress = True
+        mesh_to_egress_policy.selected_peers = selected_peers
+        return mesh_to_egress_policy
+
+    def create_deny_mesh_to_ext_policy(self):
+        """
+        Create policy for representation of the denied connections from mesh to DNS nodes
+        :return: the resulting GatewayPolicy
+        """
+        dns_peers = self.peer_container.get_all_dns_entries()
+        if not dns_peers:
+            # This is not an egress flow
+            return None
```

Comment:
Why is this the criteria for an egress flow?
Shouldn't the `deny` rules be added only for the relevant egress gateways and their external destinations?

Reply:
Refined, as described above.